### ISSUE TITLE
wix v4 -32bit / 64bit MSI creation too almost too easy

### ISSUE THREAD

**Issue by chrpai** [wixtoolset/issues#7003]
So in wix v4 switching between 32bit and 64bit MSI building is really easy.  I fear it's too easy.

I fear that if a customer accidentally changes the build settings their MSI will flip from 32 to 64 and bad things would happen.

My suggestion is an optional element or attribute in the schema that can be used to assert what platforms the author intended to support.   If the platform doesn't match this at build time throw an error asking for confirmation of the intended platform.

.NET can have this problem too when dealing with things like registry access.  In .NET there are ways to detect what bitness you are and account for that but I don't know if such things in MSI therefore I was thinking stop it at build time.

**Comment by robmen** [comment]
Setting the platform at the build level matches all the other build tools behavior. This exact behavior was possible in WiX v3, although Bundles ignored it (Burn was always 32-bit) and there could be unexpected interactions with Package/@Platform (which couldn't be removed because it was a breaking change).

WiX v4 simplifies all of this and actually builds the platform requested now. If you really need to block accidental platforms then you can use the `$(sys.BUILDARCH)` preprocessor variable.

**Comment by glen-nicol** [comment]
Ironically, I found it harder than expected to get command line to build x64. At first I assumed if the wixproj had this specified
```xml
 <Platforms>x64</Platforms>
 <RuntimeIdentifiers>win-x64</RuntimeIdentifiers>
```
 then the dotnet SDK would choose its only logical option during ```dotnet build```. I was wrong about that, AnyCPU!

Then I tried ```dotnet build --arch x64``` no luck. ```dotnet build -r win-x64```? No good.

But this worked: ```dotnet build -p:Platform=x64```. 

I admit I have gotten a little lost in the dotnet publish details as they have changed ever so slightly over the last few years but I kinda expected the SDK to set Platform based on the arch/runtime identifier.
